File: src/main/java/com/history/tree/model/Relationship.java
package com.history.tree.model;

import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

@Getter
@Setter
@Table("relationship")
public class Relationship {

    @Id
    @Column("id")
    private Long id;

    @Column("marriage_id")
    private Long marriageId;

    @Column("person_id")
    private Long personId;

    @Column("relation_person_id")
    private Long relationPersonId;

    @Column("relationship_type")
    private int relationshipType;

}

File: src/main/kotlin/com/history/tree/model/Relationship.kt
package com.history.tree.model

import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Version
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import java.util.*

@Table("relationship")
data class Relationship(

    @Id
    @Column("id")
    val id: UUID,

    @Column("marriage_id")
    val marriageId: UUID? = null,

    @Column("person_id")
    val personId: UUID,

    @Column("related_person_id")
    val relatedPersonId: UUID,

    @Column("relationship_type")
    val relationshipType: RelationshipType,

    @Column("tree_id")
    var treeId: UUID,
)